contrib/raftexample/raft.go

```diff
@@ -247,7 +302,79 @@ func (rc *raftNode) stopHTTP() {
 	<-rc.httpdonec
 }
 
+func (rc *raftNode) applySnapshot(snapshotToSave raftpb.Snapshot) {
+	if raft.IsEmptySnap(snapshotToSave) {
+		return
+	}
+
+	log.Printf("applying snapshot at index %d", rc.snapshotIndex)
+	defer log.Printf("finished applying snapshot at index %d", rc.snapshotIndex)
+
+	if snapshotToSave.Metadata.Index <= rc.appliedIndex {
+		log.Fatalf("snapshot index [%d] should > progress.appliedIndex [%d] + 1", snapshotToSave.Metadata.Index, rc.appliedIndex)
+	}
+
+	dbFilePath, err := rc.snapshotter.DBFilePath(snapshotToSave.Metadata.Index)
```

Comment:
we should not use this function. this is for etcd database file which is an attachment to the snapshot.